Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs

```diff
@@ -58,6 +60,7 @@ public MapViewModel(Xamarin.Forms.Maps.Map map, IGeolocation geolocation, INavig
                 async () => await OpenMapswithPin());
 
             PropertyChanged += DeviceChanged;
+            devicesStore.DevicesChanged += UpdateDisplayedDevice;
```

Comment:
This should probably be done in OnAppearing() and it should be unregistered in OnDiappearing(), just like CheckIfSelectedDeviceReachable. Otherwise, more and more MapViewModel-Objects will be registered but never un-registered and therefore also never garbage-collected.

Reply:
The viewmodel objects should only be created once (otherwise we would not have had the "pin remains after deletion" bug in the first place. But I will move it to the on appearing method to be consistens with the other viewmodels. 👍🏼